production/ksonnet/loki/gateway.libsonnet

```diff
@@ -46,19 +46,33 @@
             proxy_set_header     X-Scope-OrgID 1;
 
             location = /api/prom/push {
-              proxy_pass      http://distributor.%(namespace)s.svc.cluster.local$request_uri;
+              proxy_pass       http://distributor.%(namespace)s.svc.cluster.local$request_uri;
+            }
+
+            location = /api/prom/tail {
+              proxy_pass       http://querier.%(namespace)s.svc.cluster.local$request_uri;
+              proxy_set_header X-Scope-OrgID 1;
+              proxy_set_header Upgrade $http_upgrade;
+              proxy_set_header Connection "upgrade";
             }
 
             location ~ /api/prom/.* {
-              proxy_pass      http://query-frontend.%(namespace)s.svc.cluster.local$request_uri;
+              proxy_pass       http://query-frontend.%(namespace)s.svc.cluster.local$request_uri;
             }
 
             location = /loki/api/v1/push {
-              proxy_pass      http://distributor.%(namespace)s.svc.cluster.local$request_uri;
+              proxy_pass       http://distributor.%(namespace)s.svc.cluster.local$request_uri;
+            }
+
+            location = /loki/api/v1/tail {
+              proxy_pass       http://querier.%(namespace)s.svc.cluster.local$request_uri;
+              proxy_set_header X-Scope-OrgID 1;
```

Comment:
Hey, thanks for the contribution! Can you help me understand why we're setting the `X-Scope-OrgID` only in the tailing paths?

Reply:
This looks good with the exception of forcing the `X-Scope-OrgID` header as @owen-d  mentioned, we wouldn't want to do that

Reply:
To inherit the following ```X-Scope-OrgID```.
https://github.com/grafana/loki/blob/master/production/ksonnet/loki/gateway.libsonnet#L46

```
These directives are inherited from the previous level 
if and only if there are no proxy_set_header directives defined on the current level.
```
http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header

Reply:
I'm still not sure why we would want this? This feels like a strange hidden behavior if you are using this nginx proxy which fakes multi-tennancy but forces just one ID?  Loki already supports running in single-tennancy to ignore this header so I don't see why you would want this?

Reply:
It's certainly strange.
I think it would be better to pass the header field of the original request to the proxied server.
```
location = /api/prom/tail {
  proxy_pass       http://querier.%(namespace)s.svc.cluster.local$request_uri;
  proxy_pass_request_headers on;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "upgrade";
}
```

Reply:
Yes, however this is already the default and I do not believe needs to be explicitly provided:

http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass_request_headers

Reply:
I see. So I should delete ```proxy_set_header X-Scope-OrgID 1;```

Reply:
Yes please, then I think this will be all set!

Reply:
Let me fix and test tommorow 😄

Reply:
I fixed. https://github.com/grafana/loki/pull/1689/commits/54cd0eb6d06fd5f75aa92857c80901010a42f501